lib/src/lightclient.rs

```diff
@@ -1700,30 +1693,47 @@ impl LightClient {
         .map(|r| r.map_err(|e| format!("{}", e))?)
         .collect::<Result<(), String>>()?;
 
-        let (verified, heighest_tree) = verify_handle.await.map_err(|e| e.to_string())?;
-        info!("Sapling tree verification {}", verified);
+        let verify_handle =
+            tokio::spawn(async move { block_data.read().await.block_data.verify_trees().await });
+        let (verified, highest_tree) = verify_handle.await.map_err(|e| e.to_string())?;
+        info!("tree verification {}", verified);
+        info!("highest tree exists: {}", highest_tree.is_some());
         if !verified {
-            return Err("Sapling Tree Verification Failed".to_string());
+            return Err("Tree Verification Failed".to_string());
         }
 
         info!("Sync finished, doing post-processing");
 
+        let blaze_sync_data = bsync_data.read().await;
         // Post sync, we have to do a bunch of stuff
         // 1. Get the last 100 blocks and store it into the wallet, needed for future re-orgs
-        let blocks = bsync_data
-            .read()
-            .await
+        let blocks = blaze_sync_data
             .block_data
             .drain_existingblocks_into_blocks_with_truncation(MAX_REORG)
             .await;
         self.wallet.set_blocks(blocks).await;
 
+        // Store the ten highest orchard anchors
+        {
+            let mut anchors = self.wallet.orchard_anchors.write().await;
```

Comment:
Let's figure out a more accurate name for `self.wallet.orchard_anchors` and derived variables, thereof?